smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -470,6 +474,27 @@ contract MixinPurchase is
     _payProtocol(keyPrice);
   }
 
+  /**
+   * Renew a given token
+   * get referrer that was set during the original purchase
+   * @notice only works for non-free, expiring, ERC20 locks
+   * @param _tokenId the ID fo the token to renew
+   */
+  function renewMembershipFor(uint _tokenId) public {
+    _renewMembershipFor(_tokenId, _renewalConditions[_tokenId].referrer);
+  }
+
+  /**
+   * Renew a given token
+   * get referrer that was set during the original purchase
+   * @notice only works for non-free, expiring, ERC20 locks
+   * @param _tokenId the ID fo the token to renew
+   * @param _referrer the address of the referrer
+   */
+  function renewMembershipFor(uint _tokenId, address _referrer) public {
+    _renewMembershipFor(_tokenId, _referrer);
+  }
+
```

Comment:
not sure we really need both signature. We can just reuse the existing referrer when zero address is passed?

Reply:
I think it is better for backwards compact? I would make this one as deprecated .. and do that change:
```suggestion
  /**
   * Renew a given token
   * get referrer that was set during the original purchase
   * @notice only works for non-free, expiring, ERC20 locks
   * @param _tokenId the ID fo the token to renew
   */
  function renewMembershipFor(uint _tokenId) public {
    _renewMembershipFor(_tokenId, _renewalConditions[_tokenId].referrer);
  }

  /**
   * Renew a given token
   * get referrer that was set during the original purchase
   * @notice only works for non-free, expiring, ERC20 locks
   * @param _tokenId the ID fo the token to renew
   * @param _referrer the address of the referrer
   */
  function renewMembershipFor(uint _tokenId, address _referrer) public {
    _renewMembershipFor(_tokenId, _renewalConditions[_tokenId].referrer);
  }

```

Reply:
a simpler way maybe is to not add a new signature. If a referrer has been specified earlier, it takes precedent. if not, it takes the referrer from function args. wdyt ?